packages/app/src/device-file-system/commands/retrieve-files.command.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { Endpoint, Method } from "@mudita/pure"
+import { AppError } from "App/core/errors"
+import { Result, ResultObject } from "App/core/builder"
+import { RequestResponseStatus } from "App/core/types/request-response.interface"
+import { BaseCommand } from "App/device-file-system/commands/base.command"
+import { DeviceFileSystemError } from "App/device-file-system/constants"
+
+export class RetrieveFilesCommand extends BaseCommand {
+  public async exec(directory: string): Promise<
+    ResultObject<
+      | Record<
+          string,
+          {
+            path: string
+            fileSize: number
+          }[]
+        >
+      | undefined
+    >
+  > {
+    const { data, status, error } = await this.deviceService.request({
+      endpoint: Endpoint.FileSystem,
+      method: Method.Get,
+      body: {
+        listDir: directory,
+      },
+    })
+
+    if (status !== RequestResponseStatus.Ok || !data) {
+      return Result.failed(
+        new AppError(
+          DeviceFileSystemError.FilesRetrieve,
+          error ? error.message : "Something wen't wrong"
```

Comment:
```suggestion
          error ? error.message : "Something went wrong"
```

Reply:
I'll fix it in the next PR

Reply:
https://github.com/mudita/mudita-center/pull/906/commits/2c7dd2e1e48fd486028de7c64e2cf57e6fa08281